frontend/src/component/common/ConstraintAccordion/ConstraintAccordionEdit/ConstraintAccordionEditBody/DateSingleValue/DateSingleValue.tsx

```diff
@@ -1,42 +1,93 @@
+
+interface ITimezoneSelect extends ISelectOption {
+    utcOffset: string;
+}
 interface IDateSingleValueProps {
     setValue: (value: string) => void;
     value?: string;
     error: string;
     setError: React.Dispatch<React.SetStateAction<string>>;
 }
 
+const StyledWrapper = styled('div')(({ theme }) => ({
+    display: 'flex',
+    flexDirection: 'row',
+    marginBottom: theme.spacing(1),
+    alignItems: 'center',
+    gap: theme.spacing(1),
+}));
+
 export const DateSingleValue = ({
     setValue,
     value,
     error,
     setError,
 }: IDateSingleValueProps) => {
+    const timezones = Object.values(
+        TimezoneCountries.getAllTimezones({ deprecated: false })
+    ).map(timezone => ({
+        key: timezone.name,
+        label: `${timezone.name}`,
+        utcOffset: timezone.utcOffsetStr,
+    }));
+    const { timeZone: localTimezoneName } =
+        Intl.DateTimeFormat().resolvedOptions();
+    const [pickedDate, setPickedDate] = useState(value || '');
+
+    const localTimezone = useMemo<ITimezoneSelect | undefined>(() => {
+        return timezones.find(t => t.key === localTimezoneName);
+    }, [timezones, localTimezoneName]);
+
     if (!value) return null;
 
     return (
         <>
             <ConstraintFormHeader>Select a date</ConstraintFormHeader>
-            <Input
-                id="date"
-                label="Date"
-                type="datetime-local"
-                value={parseDateValue(value)}
-                onChange={e => {
-                    setError('');
-                    const parsedDate = parseValidDate(e.target.value);
-                    const dateString = parsedDate?.toISOString();
-                    dateString && setValue(dateString);
-                }}
-                InputLabelProps={{
-                    shrink: true,
-                }}
-                error={Boolean(error)}
-                errorText={error}
-                required
-            />
+            <StyledWrapper>
+                <Input
+                    id="date"
+                    label="Date"
+                    type="datetime-local"
+                    value={parseDateValue(pickedDate)}
+                    onChange={e => {
+                        setError('');
+                        const parsedDate = parseValidDate(e.target.value);
+                        const dateString = parsedDate?.toISOString();
+                        dateString && setPickedDate(dateString);
+                        dateString && setValue(dateString);
+                    }}
+                    InputLabelProps={{
+                        shrink: true,
+                    }}
+                    error={Boolean(error)}
+                    errorText={error}
+                    required
+                />
+                <ConditionallyRender
+                    condition={Boolean(localTimezone)}
+                    show={
+                        <p>
+                            {localTimezone!.key}{' '}
```

Comment:
I don't usually like the `!` operator, but in this case it is safe since we already check for truthiness

Reply:
Yes, I agree. However, it does introduce a potential for error later if the condition were to change. Could this perhaps be changed to regular ternary? E.g.

```
Boolean(localTimezone) ? <p>{localTimezone.key[...]</p> : <p>The time shown is your browser's local time</p>
```

Maybe even extract the string to a variable higher up?

```
const timeZoneDescription = localTimezone ? `${localTimezone.key} [...]` : "The time shown is your browser's local time"
```

and then skip the last conditional altogether?

Reply:
If the condition suddenly changes to `condition={true}` (or any other condition that doesn't rely on `localTimezone` being present), then the `!` will try and access an object that is undefined.

Reply:
Or, well, it could 💁🏼 So I think if we can avoid introducing a potential error like that, then we should.